test/functional/p2p_invalid_messages.py

```diff
@@ -125,6 +128,16 @@ def test_msgtype(self):
             conn.sync_with_ping(timeout=1)
         self.nodes[0].disconnect_p2ps()
 
+    def test_inv_msg_with_negative_type(self):
+        self.log.info("Test inv with negative type raises argument out of range error")
+        invalid_inv = msg_inv([CInv(t=-1, h=1)])  # inv type must be an unsigned integer
+        try:
+            self.nodes[0].add_p2p_connection(P2PInterface()).send_message(invalid_inv)
+        except struct.error as e:
```

Comment:
We generally don't test the test framework. Sending an uint32_t{-1} and observing the node doesn't crash with sanitizers enabled should be a sufficient smoke test. Maybe add an assert_debug_log to observe how the inv will be formatted.

```
            LogPrint(BCLog::NET, "got inv: %s  %s peer=%d\n", inv.ToString(), fAlreadyHave ? "have" : "new", pfrom.GetId());

Reply:
Thanks for the tip. The temporary framework test helped me see an issue with the framework update, will drop it and try your suggestion.

Reply:
Thanks Marco. Trying some stuff to see how the CI reacts to it.